src/msg/async/frames_v2.h

```diff
@@ -55,7 +56,9 @@ enum class Tag : __u8 {
   MESSAGE,
   KEEPALIVE2,
   KEEPALIVE2_ACK,
-  ACK
+  ACK,
+  COMPRESSION_REQUEST,
```

Comment:
could you explain a little bit why compression negotiation can not be piggy-backed to existing negotiation process of msgr v2?

Reply:
The existing msgr 2.0 protocol first handles supported/required protocol features declaration and then starts the auth negotiation.
The first phase determines which features should be used in the protocol. It is only meant to define the flow the client/server will follow, therefore we cannot piggyback there.
This leaves us with the second phase (auth negotiation) -
The auth negotiation is a bit more complex than compression. It can result in several messages being sent between peers (see [authentication](https://docs.ceph.com/en/latest/dev/msgr2/#authentication)). 
Piggybacking compression into the auth handshake will require a massive duplication of these messages including complexity to the code. 
I also think that the flexibility of the msgr 2.0 would be compromised. Piggybacking compression opens the door to piggybacking future features as well. This means that during handshake we might pass redundant data and increase our network usage when we could have avoided it. In addition, compression is a pretty simple case. Other features may result in additional messages until a resolution is made.